README.md

```diff
@@ -1,481 +1,80 @@
+  - [Vault](docs/ADMIN.md#vault)
+  - [Running](docs/ADMIN.md#running)
+- [User Guide](docs/USER.md)
+- [Contributing](docs/DEVELOP.md)
+  - [Building & Testing](docs/DEVELOP.md#building---testing)
+      * [KCP](docs/DEVELOP.md#kcp)
+      * [Out of cluster](docs/DEVELOP.md#out-of-cluster)
+      * [In cluster](docs/DEVELOP.md#in-cluster)
+  - [Debugging](docs/DEVELOP.md#debugging)
+  - [Manual testing with custom images](docs/DEVELOP.md#manual-testing-with-custom-images)
+      * [Requirements on the Service Providers](docs/DEVELOP.md#requirements-on-the-service-providers)
+      * [To test SPI on CPS/KCP cluster](docs/DEVELOP.md#to-test-spi-on-cps-kcp-cluster)
+- [License](LICENSE)
+
+## About
+The Service Provider Integration aims to provide a service-provider-neutral way (as much as reasonable) of obtaining authentication tokens so that tools accessing the service provider do not have to deal with the intricacies of obtaining the access tokens from the different service providers.
+The caller creates an `SPIAccessTokenBinding` object in a Kubernetes namespace where they specify the repository to which they require access and the required permissions
+in the repo. The caller also specifies the target `Secret` to which they want the access token to the repository to be persisted.
+The caller then watches the `SPIAccessTokenBinding` and reacts to its status changing by either initiating the provided OAuth flow or using the created Secret object.
+Additionally, SPI provides an HTTP endpoint for manually uploading the access token for a certain `SPIAccessToken` object. Therefore,
 the user doesn’t have to go through OAuth flow for tokens that they manually provide the access token for.
 
-### High level view
+### Architecture
 There are 2 main components. SPI HTTP API which is required for parts of the workflow that require direct user interaction and SPI CRDs (and a controller manager for them).
 The custom resources are meant to be used by the eventual Consumers of the secrets that require access tokens to communicate with service providers.
 Therefore, the main audience of SPI is the Consumers of the secrets. In the case of App Studio, this is most probably going to be HAS and/or HAC.
-Because SPI requires user interaction for a part of its functionality, HAC will need to interface with the SPI REST API so that authenticated requests can be made to 
+Because SPI requires user interaction for a part of its functionality, HAC will need to interface with the SPI REST API so that authenticated requests can be made to
 the cluster and the service providers.
-SPI will not provide any user-facing UI on its own. 
+SPI will not provide any user-facing UI on its own.
 Instead, for stuff that will need user interaction, it will provide links to its REST API that will either consume supplied data or redirect to a service provider (in case of OAuth flow).
 
+## Service Provider Integration Glossary
```

Comment:
this same glossary is in `USER.md`, I would choose just one place so they won't get out of sync.